lib/auth/in_memory/index.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * Class that provides an internal indexing over the simple data provided by
+ * the authentication configuration file for the memory backend. This allows
+ * accessing the different authentication entities through various types of
+ * keys.
+ *
+ * @class Index
+ */
+export default class Index {
```

Comment:
`Indexer` would be more appropriate here, as conventionally `index.js` just means root file for the project.